pkg/cluster/install.go

```diff
@@ -56,8 +56,15 @@ func (m *manager) AdminUpdate(ctx context.Context) error {
 
 func (m *manager) Update(ctx context.Context) error {
 	steps := []steps.Step{
-		steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.validateResources)),
+		steps.Action(m.initializeKubernetesClients), // must be first
 		steps.Action(m.initializeClusterSPClients),
+		steps.Action(m.clusterSPObjectID),
+		steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.validateResources)),
+		// credentials rotation flow steps
+		steps.Action(m.createOrUpdateClusterServicePrincipalRBAC),
+		steps.Action(m.createOrUpdateDenyAssignment),
+		steps.Action(m.updateAROSecret),
```

Comment:
nit: moving this down would mean we'd be less likely to be in-flight during an api server rotation, fwiw